src/components/auth/login/Recaptcha/index.js

```diff
@@ -10,46 +10,57 @@ const log = logger.child({ from: 'init' })
 const { recaptchaSiteKey, publicUrl } = Config
 
 const Recaptcha = React.forwardRef(({ onSuccess = noop, onFailure = noop, children }, ref) => {
-  const isPassedRef = useRef(false)
   const captchaRef = useRef()
+  const isReadyRef = useRef()
+  const setReadyRef = useRef()
+  const isPassedRef = useRef(false)
+  const onCaptchaLoaded = useCallback(() => setReadyRef.current(), [])
 
   const onStatusChange = useCallback(
-    async result => {
-      log.debug('Recaptcha result', result)
-      let hasPassed
+    async payload => {
+      log.debug('Recaptcha payload', payload)
+
       try {
-        const res = await API.verifyCaptcha(result)
-        log.debug('Recaptcha verify res', res)
-
-        if (res.data.success) {
-          hasPassed = true
-        } else {
-          hasPassed = false
-        }
+        const result = await API.verifyCaptcha(payload)
+
+        log.debug('Recaptcha verify result', result)
+        isPassedRef.current = get(result, 'data.success', false)
       } catch (exception) {
-        log.error('recaptcha verification failed', exception.message, exception, result)
-        hasPassed = false
+        log.error('recaptcha verification failed', exception.message, exception, payload)
       } finally {
-        isPassedRef.current = hasPassed
-        ;(hasPassed ? onSuccess : onFailure)()
+        ;(isPassedRef.current ? onSuccess : onFailure)()
       }
     },
     [onSuccess, onFailure],
   )
 
   useImperativeHandle(ref, () => ({
     hasPassedCheck: () => isPassedRef.current,
-    launchCheck: () => {
+    launchCheck: async () => {
       if (isPassedRef.current) {
         return
       }
 
+      await isReadyRef.current
       captchaRef.current.launch()
     },
   }))
+  ;(() => {
+    if (isReadyRef.current) {
+      return
+    }
+
+    isReadyRef.current = new Promise(resolve => (setReadyRef.current = resolve))
+  })()
```

Comment:
Wouldn't it be better to place this inside an `useEffect` hook with isReadyRef.current in the dependency array?